app/src/main/kotlin/io/privkey/keep/MnemonicRecoveryScreen.kt

```
@@ -0,0 +1,379 @@
+                    mnemonicData.update(mnemonic)
+                    try {
+                        val cipher = onGetCipher()
+                        onBiometricAuth(cipher) { authedCipher ->
+                            if (authedCipher != null) {
+                                onCreateAccount(mnemonicData.valueUnsafe(), "", name, authedCipher)
+                            }
+                        }
+                    } catch (e: KeyPermanentlyInvalidatedException) {
+                        if (BuildConfig.DEBUG) Log.e("MnemonicRecovery", "Biometric key invalidated: ${e::class.simpleName}")
+                        onError("Biometric key invalidated. Please re-enroll biometrics.")
+                    } catch (e: BiometricHelper.BiometricNotReadyException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)